src/main/kotlin/blaster707/contactlist/ContactList.kt

```diff
@@ -13,14 +13,22 @@ data class ContactList(val contacts: MutableList<ContactEntry>) {
         return output
     }
 
-    fun deleteContactEntry(contactToDelete: String): Boolean {
+    fun contactEntryIdLocated(contactIdtoLocate: String): Boolean {

```

Comment:
I like that you've broken the "find" functionality apart from the delete or update functionality - good work using functions to compose smaller functions together into larger functions.

Few suggestions:

- When building functions that return `Boolean`, it's customary, although not required, to name them in the form of a terse question. Rather than a statement like `contactEntryIdLocated`, I might change it to something like `doesContactExist` or `doesContactEntryExist`. Think of it like you are asking the function a question and it's responding with an answer.  Additionally, I wouldn't put `Id` in the name, since "no one cares" if the ID exists or is located - they just want to know if the contact exists.  The Id is just the way we refer to which contact we want to know if it exists.  Subtle, yes, but over time these subtleties can help or hinder the readability of the code.

- Rather than asking if a contact exists, maybe build a `fun findContact(id: String): ContactEntry?` function, that returns the actual `ContractEntry` if it is found, or `null` if it is not.  That way you don't have to do the "locating" work multiple times.

- You might consider using [`firstOrNull`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first-or-null.html) or [`singleOrNull`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/single-or-null.html) to do the actual searching work, rather than using a for-loop. Similar to [`map`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map.html) and [`filter`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter.html), finding stuff in a collection is so common that it's been put into helper functions like `first`, `firstOrNull`, `last`, `lastOrNull`, `single`, `singleOrNull`, etc.